source/AI.cpp

```diff
@@ -386,12 +396,26 @@ void AI::UpdateKeys(PlayerInfo &player, Command &activeCommands)
 		IssueDeploy(player);
 
 	shared_ptr<Ship> target = flagship->GetTargetShip();
+	shared_ptr<Minable> targetAsteroid = flagship->GetTargetAsteroid();
 	Orders newOrders;
-	if(activeCommands.Has(Command::FIGHT) && target && !target->IsYours())
+	if(activeCommands.Has(Command::FIGHT) && ((target && !target->IsYours()) || targetAsteroid))
```

Comment:
Could this result in being able to focus fire your own ships should you have an asteroidTarget as well? I'm not sure if the target is reset instantly when your target asteroid has been destroyed. Maybe there is some way I'm not thinking of to keep an asteroid target, I am not sure if getting a new ship target clears the asteroidTarget, I don't think so.